### PR TITLE
a bunch of tests handling edge cases

### PR DESCRIPTION
**Pull request by EneOli**
Added a bunch of my own tests handling edge cases. Checks many network operations,

### CODE REVIEW

File: src/test/java/edu/kit/informatik/eneoli/ip/EneOliIPTest.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        Assertions.assertFalse(ip3.equals(ip1));
+    }
+
+    @Test
+    public void addressOrder() throws ParseException {
+        IP ip1 = new IP("1.255.255.255");
+        IP ip2 = new IP("100.255.255.255");
+        IP ip3 = new IP("1.255.255.42");
+
+        List<IP> list = new ArrayList<IP>(Arrays.asList(ip1, ip2, ip3));
```
Comment (janik6882) [inline src/test/java/edu/kit/informatik/eneoli/ip/EneOliIPTest.java:96]: List import missing

Comment (EneOli) [reply src/test/java/edu/kit/informatik/eneoli/ip/EneOliIPTest.java:98]: fixed

File: src/test/java/edu/kit/informatik/eneoli/EneOliNetworkTest.java [hunk 2]
```diff
@@ -0,0 +1,1066 @@
...
+        Assertions.assertEquals("4.4.4.4", level3.get(0).toString());
+
+        // LEVEL 4
+        Assertions.assertEquals(2, level4.size());
+        Assertions.assertEquals("5.5.5.5", level4.get(0).toString());
+        Assertions.assertEquals("6.6.6.6", level4.get(1).toString());
+
+        // EXTRA
+
+        Assertions.assertEquals(1, network.getTopologies().size());
```
[the remark this answers is not in the document]
Comment (EneOli) [reply src/test/java/edu/kit/informatik/eneoli/EneOliNetworkTest.java]: fixed

File: src/test/java/edu/kit/informatik/eneoli/EneOliNetworkTest.java [hunk 3]
```diff
@@ -0,0 +1,1066 @@
...
+
+    @Test
+    public void listConstructorSimpleTest() throws ParseException {
+        Network network = new Network(new IP("1.1.1.1"), Arrays.asList(
+                new IP("2.2.2.2"),
+                new IP("3.3.3.3"),
+                new IP("4.4.4.4")
+        ));
+
+        List<IP> ips = network.list();
```
[the remark this answers is not in the document]
Comment (EneOli) [reply src/test/java/edu/kit/informatik/eneoli/EneOliNetworkTest.java:25]: fixed

**Review by janik6882 — CHANGES_REQUESTED** [review]
Importing of IP, Network and ParseException not existing. Maybe solve by importing KoeriTestUtils?

**Comment by EneOli** [comment]
May help, we need more information about how the Test framework loads the classes
